light/client.go

```diff
@@ -1117,6 +1117,11 @@ func (c *Client) findNewPrimary(ctx context.Context, height int64, remove bool)
 		}
 	}
 
+	// remove witnesses marked as bad. Removal is done in descending order
+	if err := c.removeWitnesses(witnessesToRemove); err != nil {
+		return nil, err
+	}
+
```

Comment:
It looks like the code was intended to return the last error recovered by the above loop, but here we supersede that. I'm not sure if it matters, but if it does, consider intead:
```suggestion
	if err := c.removeWitnesses(witnessesToRemove); err != nil && lastError == nil {
		lastError = err
	}

```

note that in either case we'll return a nil result anyway, but this preserves the error semantics if we care. We might also consider logging it.

Reply:
Yeah I think it might be better to log the error